service/src/main/kotlin/fi/espoo/evaka/shared/domain/Time.kt

```diff
@@ -290,7 +290,8 @@ fun asDistinctPeriods(periods: List<DateRange>, spanningPeriod: DateRange): List
             .sortedBy { orMax(it) }
 
     return allStartDates
-        .map { start -> DateRange(start, allEndDates.find { end -> start <= orMax(end) }) }
+        // TODO: !! added, might fail?
```

Comment:
does this comment need resolving?

Reply:
Comment removed. I don't think it needs resolving because for all start dates there should be the next end date as we're dealing with finite date ranges.